### ISSUE TITLE
Frequent GPF in logs

### ISSUE THREAD

**Issue by jimko** [TheAssassin/AppImageLauncher#558]
### Pre-submit checks

- [X] I [checked for similar issues](https://github.com/TheAssassin/AppImageLauncher/issues?q=) beforehand, but could not find any, not even closed ones. I could not add my bug report to any existing issue.
- [X] I am going to take the time to to fill in all the required details. I know that the bug report will be dismissed otherwise.

### Describe the bug

I see the following in my dmesg output every couple of seconds.  Researching a USB issue and found a thousand of these

```
#dmesg
[45008.873994] traps: appimagelaunche[24764] general protection fault ip:7f01b82c6664 sp:7ffe9158ded8 error:0 in libstdc++.so.6.0.30[7f01b829a000+110000]
```
Here's the whole cycle from syslog which repeats every minute
```
Feb  7 09:55:53 cantina appimagelauncherd[25396]: Searching for existing AppImages
Feb  7 09:55:53 cantina appimagelauncherd[25396]: Searching directory: /home/jimko/Applications
Feb  7 09:55:53 cantina appimagelauncherd[25396]: Found AppImage: /home/jimko/Applications/balenaEtcher-1.10.6-x64_07316bb894b47a07b828670221e47704.AppImage
Feb  7 09:55:53 cantina appimagelauncherd[25396]: AppImage integrated already, skipping
Feb  7 09:55:53 cantina appimagelauncherd[25396]: Found AppImage: /home/jimko/Applications/arduino-ide_2.0.3_Linux_64bit.AppImage
Feb  7 09:55:53 cantina appimagelauncherd[25396]: AppImage integrated already, skipping
Feb  7 09:55:53 cantina appimagelauncherd[25396]: Found AppImage: /home/jimko/Applications/Ultimaker-Cura-5.2.1-linux.AppImage
Feb  7 09:55:53 cantina appimagelauncherd[25396]: AppImage integrated already, skipping
Feb  7 09:55:53 cantina appimagelauncherd[25396]: Searching directory: /home/jimko/Downloads
Feb  7 09:55:53 cantina appimagelauncherd[25396]: Unknown ELF data order 0
Feb  7 09:55:53 cantina appimagelauncherd[25396]: Watching directories: /home/jimko/Applications /home/jimko/Downloads
Feb  7 09:56:23 cantina systemd[1962]: appimagelauncherd.service: Main process exited, code=killed, status=11/SEGV
Feb  7 09:56:23 cantina systemd[1962]: appimagelauncherd.service: Failed with result 'signal'.
Feb  7 09:56:23 cantina kernel: [46048.872296] traps: appimagelaunche[25396] general protection fault ip:7f7499ec6664 sp:7ffefec21498 error:0 in libstdc++.so.6.0.30[7f7499e9a000+110000]
Feb  7 09:56:33 cantina systemd[1962]: appimagelauncherd.service: Scheduled restart job, restart counter is at 1151.

```

### Expected behavior

Not expecting to see GPF every time.  

### Steps to reproduce the issue

dmesg | grep "appimagelaunche"

### Distribution and desktop environment

uname -a
Linux cantina 5.19.0-29-generic #30-Ubuntu SMP PREEMPT_DYNAMIC Wed Jan 4 12:14:09 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

cat /etc/*release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.10
DISTRIB_CODENAME=kinetic
DISTRIB_DESCRIPTION="Ubuntu 22.10"
PRETTY_NAME="Ubuntu 22.10"
NAME="Ubuntu"
VERSION_ID="22.10"
VERSION="22.10 (Kinetic Kudu)"
VERSION_CODENAME=kinetic
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=kinetic
LOGO=ubuntu-logo

### Installed AppImageLauncher version

AppImageLauncher --appimagelauncher-version
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
AppImageLauncher version 2.2.0 (git commit 0f91801), built on 2022-11-02 23:51:55 UTC

### List of AppImages you tried

arduino-ide_2.0.3_Linux_64bit (works)
Ultimaker-Cura-5.2.1-linux (works)
balenaEtcher-1.10.6-x64_07316bb894b47a07b828670221e47704 (works)

### Additional context

Not really causing a problem for me. Just noticed because I couldn't get an arduino board to be recognized on USB and checked dmesg.

**Comment by azuredusk10** [comment]
This is also happening for me. Every time I log into my system on Fedora 38, I get between 1 - 3 notifications to say the system encountered a problem.

In the "Details" window of "Problem reporting", it says this:
`traps: appimagelaunche[5561] general protection fault ip:7f2c488cd444 sp:7ffc2124c098 error:0 in libstdc++.so.6.0.32[7f2c4889c000+12a000]`

**Comment by trinitronx** [comment]
Confirmed on Manjaro Linux:

```bash
$ uname -r
6.1.31-2-MANJARO

$ sudo dmesg -H
[Jun23 20:40] microcode: microcode updated early to revision 0xf2, date = 2022-12-26
[  +0.000000] Linux version 6.1.31-2-MANJARO (builduser@fv-az462-57) (gcc (GCC) 13.1.1 20230429, GNU ld (GNU Binutils) 2.40.0) #1 SMP PREEMPT_DYNAMIC Sun Jun  4 12:31:46 UTC 2023
[  +0.000000] Command line: BOOT_IMAGE=/boot/vmlinuz-6.1-x86_64 root=UUID=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx rw quiet apparmor=1 security=apparmor udev.log_priority=3
[... SNIP ...]
[Jun23 20:41] show_signal: 50 callbacks suppressed
[  +0.000006] traps: appimagelaunche[863] general protection fault ip:7efceacca394 sp:7ffd3e0da728 error:0 in libstdc++.so.6.0.31[7efceac9c000+137000]
[ +39.999983] traps: appimagelaunche[1733] general protection fault ip:7fce588ca394 sp:7ffdd5aada68 error:0 in libstdc++.so.6.0.31[7fce5889c000+137000]
[Jun23 20:42] traps: appimagelaunche[1774] general protection fault ip:7fbb038ca394 sp:7ffc24914b98 error:0 in libstdc++.so.6.0.31[7fbb0389c000+137000]
[Jun23 20:43] traps: appimagelaunche[1798] general protection fault ip:7fd6f66ca394 sp:7fff85018258 error:0 in libstdc++.so.6.0.31[7fd6f669c000+137000]
[ +32.888131] snd_hda_codec_hdmi hdaudioC1D0: HDMI: audio coding type 0 not expected
[  +7.110736] traps: appimagelaunche[1918] general protection fault ip:7f09a12ca394 sp:7fffb3537bf8 error:0 in libstdc++.so.6.0.31[7f09a129c000+137000]
[Jun23 20:44] snd_hda_codec_hdmi hdaudioC1D0: HDMI: audio coding type 0 not expected
[ +27.960715] traps: appimagelaunche[2042] general protection fault ip:7f82958ca394 sp:7fff42dd9738 error:0 in libstdc++.so.6.0.31[7f829589c000+137000]
[Jun23 20:45] snd_hda_codec_hdmi hdaudioC1D0: HDMI: audio coding type 0 not expected
[  +5.080076] traps: appimagelaunche[2163] general protection fault ip:7f77658ca394 sp:7ffdd1d141b8 error:0 in libstdc++.so.6.0.31[7f776589c000+137000]
[  +3.922382] snd_hda_codec_hdmi hdaudioC1D0: HDMI: audio coding type 0 not expected
[ +16.600105] snd_hda_codec_hdmi hdaudioC1D0: HDMI: audio coding type 0 not expected
[ +19.476551] traps: appimagelaunche[2407] general protection fault ip:7fcd72eca394 sp:7ffc09acddc8 error:0 in libstdc++.so.6.0.31[7fcd72e9c000+137000]
[Jun23 20:46] traps: appimagelaunche[2540] general protection fault ip:7f5211eca394 sp:7ffc9bd0ef18 error:0 in libstdc++.so.6.0.31[7f5211e9c000+137000]
[Jun23 20:47] traps: appimagelaunche[2729] general protection fault ip:7ff26eeca394 sp:7ffc6df91ee8 error:0 in libstdc++.so.6.0.31[7ff26ee9c000+137000]
[ +40.000065] traps: appimagelaunche[2807] general protection fault ip:7fa2c48ca394 sp:7ffc478c9d28 error:0 in libstdc++.so.6.0.31[7fa2c489c000+137000]
[Jun23 20:48] traps: appimagelaunche[2911] general protection fault ip:7f748b8ca394 sp:7fff3fde42f8 error:0 in libstdc++.so.6.0.31[7f748b89c000+137000]
[Jun23 20:49] traps: appimagelaunche[3008] general protection fault ip:7f99d68ca394 sp:7fffbee2e0c8 error:0 in libstdc++.so.6.0.31[7f99d689c000+137000]
[ +21.126513]  sdd: sdd1 sdd2 sdd3 sdd4 sdd5 sdd6 sdd7 sdd8
[ +16.412224]  sdd: sdd1 sdd2 sdd3 sdd4 sdd5 sdd6 sdd7 sdd8
[  +2.150464]  sdf: sdf1
[  +0.309318] traps: appimagelaunche[3138] general protection fault ip:7f71324ca394 sp:7fff83da1b98 error:0 in libstdc++.so.6.0.31[7f713249c000+137000]
[Jun23 20:50] traps: appimagelaunche[3416] general protection fault ip:7f47544ca394 sp:7ffe333337b8 error:0 in libstdc++.so.6.0.31[7f475449c000+137000]
[Jun23 20:51] traps: appimagelaunche[3442] general protection fault ip:7efcaceca394 sp:7ffdddb14358 error:0 in libstdc++.so.6.0.31[7efcace9c000+137000]
[ +39.998834] traps: appimagelaunche[3552] general protection fault ip:7f2708eca394 sp:7ffe1b9c5d98 error:0 in libstdc++.so.6.0.31[7f2708e9c000+137000]
[Jun23 20:52] traps: appimagelaunche[4029] general protection fault ip:7f49c38ca394 sp:7fff2edd9358 error:0 in libstdc++.so.6.0.31[7f49c389c000+137000]
[Jun23 20:53] traps: appimagelaunche[6342] general protection fault ip:7eff946ca394 sp:7ffd1cf98408 error:0 in libstdc++.so.6.0.31[7eff9469c000+137000]
[ +39.998732] traps: appimagelaunche[6444] general protection fault ip:7f60002ca394 sp:7ffff6512438 error:0 in libstdc++.so.6.0.31[7f600029c000+137000]
[Jun23 20:54] traps: appimagelaunche[6589] general protection fault ip:7fd2a98ca394 sp:7fff81828f38 error:0 in libstdc++.so.6.0.31[7fd2a989c000+137000]
[Jun23 20:55] traps: appimagelaunche[6658] general protection fault ip:7febd00ca394 sp:7ffc045aa218 error:0 in libstdc++.so.6.0.31[7febd009c000+137000]
[ +39.998753] traps: appimagelaunche[6725] general protection fault ip:7f82980ca394 sp:7ffc5d2c44e8 error:0 in libstdc++.so.6.0.31[7f829809c000+137000]
```

Core dumps are available via `coredumpctl`:

```bash
$ coredumpctl info 863
           PID: 863 (appimagelaunche)
           UID: 1000 (trinitronx)
           GID: 1000 (trinitronx)
        Signal: 11 (SEGV)
     Timestamp: Fri 2023-06-23 20:41:09 MDT (20min ago)
  Command Line: /usr/bin/appimagelauncherd
    Executable: /usr/bin/appimagelauncherd
 Control Group: /user.slice/user-1000.slice/user@1000.service/app.slice/appimagelauncherd.service
          Unit: user@1000.service
     User Unit: appimagelauncherd.service
         Slice: user-1000.slice
     Owner UID: 1000 (trinitronx)
       Boot ID: c49162deec5648af9d446d77ca58abf9
    Machine ID: 132f3bb4c5df4c7b9bc3e7b3947f4b7a
      Hostname: saturn
       Storage: /var/lib/systemd/coredump/core.appimagelaunche.1000.c49162deec5648af9d446d77ca58abf9.863.1687574469000000.zst (present)
  Size on Disk: 759.3K
       Message: Process 863 (appimagelaunche) of user 1000 dumped core.
                
                Stack trace of thread 863:
                #0  0x00007efceacca394 local_Rb_tree_increment (libstdc++.so.6 + 0xca394)
                #1  0x000055b891da0e71 n/a (appimagelauncherd + 0x16e71)
                #2  0x000055b891d9a64a n/a (appimagelauncherd + 0x1064a)
                #3  0x00007efceb2beb41 n/a (libQt5Core.so.5 + 0x2beb41)
                #4  0x00007efceb2c109f _ZN6QTimer7timeoutENS_14QPrivateSignalE (libQt5Core.so.5 + 0x2c109f)
                #5  0x00007efceb2b1c26 _ZN7QObject5eventEP6QEvent (libQt5Core.so.5 + 0x2b1c26)
                #6  0x00007efceb28e00c _ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent (libQt5Core.so.5 + 0x28e00c)
                #7  0x00007efceb2d8583 _ZN14QTimerInfoList14activateTimersEv (libQt5Core.so.5 + 0x2d8583)
                #8  0x00007efceb2d8bc2 n/a (libQt5Core.so.5 + 0x2d8bc2)
                #9  0x00007efceb6a653b g_main_context_dispatch (libglib-2.0.so.0 + 0x5a53b)
                #10 0x00007efceb703219 n/a (libglib-2.0.so.0 + 0xb7219)
                #11 0x00007efceb6a51a2 g_main_context_iteration (libglib-2.0.so.0 + 0x591a2)
                #12 0x00007efceb2d8d3c _ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE (libQt5Core.so.5 + 0x2d8d3c)
                #13 0x00007efceb28668c _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE (libQt5Core.so.5 + 0x28668c)
                #14 0x00007efceb2912f9 _ZN16QCoreApplication4execEv (libQt5Core.so.5 + 0x2912f9)
                #15 0x000055b891d950e1 n/a (appimagelauncherd + 0xb0e1)
                #16 0x00007efceaa39850 n/a (libc.so.6 + 0x23850)
                #17 0x00007efceaa3990a __libc_start_main (libc.so.6 + 0x2390a)
                #18 0x000055b891d95b55 n/a (appimagelauncherd + 0xbb55)
                ELF object binary architecture: AMD x86-64

```

**Comment by trinitronx** [comment]
@TheAssassin :

Just wanted to revisit this because it is still happening.  It appears that Arch Linux packages `extra/appimagelauncher` version `2.2.0-10` is affected.  Same stacktrace as before with a crash in `local_rb_tree_increment`.  No debug symbols provided for this package, [and Manjaro's broken `debuginfod` implementation makes this difficult][1], sorry.

<details><summary><code>coredumpctl info</code> stacktrace</summary>
<p>

```console
$ sudo coredumpctl info 155885                                                                                    

           PID: 155885 (appimagelaunche)
           UID: 1000 (trinitronx)
           GID: 1000 (trinitronx)
        Signal: 11 (SEGV)
     Timestamp: Wed 2024-07-10 08:10:31 MDT (2h 31min ago)
  Command Line: /usr/bin/appimagelauncherd
    Executable: /usr/bin/appimagelauncherd
 Control Group: /user.slice/user-1000.slice/user@1000.service/app.slice/appimagelauncherd.service
          Unit: user@1000.service
     User Unit: appimagelauncherd.service
         Slice: user-1000.slice
     Owner UID: 1000 (trinitronx)
       Boot ID: ef624dc5b47c4d73a7fb6093c319138c
    Machine ID: 132f3bb4c5df4c7b9bc3e7b3947f4b7a
      Hostname: saturn
       Storage: /var/lib/systemd/coredump/core.appimagelaunche.1000.ef624dc5b47c4d73a7fb6093c319138c.155885.1720620631000000.zst (present)
  Size on Disk: 731.6K
       Message: Process 155885 (appimagelaunche) of user 1000 dumped core.

                Stack trace of thread 155885:
                #0  0x000073e3af0c6c04 local_Rb_tree_increment (libstdc++.so.6 + 0xc6c04)
                #1  0x000055d47aa6e909 n/a (appimagelauncherd + 0x16909)
                #2  0x000055d47aa69de7 n/a (appimagelauncherd + 0x11de7)
                #3  0x000073e3af6dfa9e n/a (libQt5Core.so.5 + 0x2dfa9e)
                #4  0x000073e3af6e1aae _ZN6QTimer7timeoutENS_14QPrivateSignalE (libQt5Core.so.5 + 0x2e1aae)
                #5  0x000073e3af6d1cc9 _ZN7QObject5eventEP6QEvent (libQt5Core.so.5 + 0x2d1cc9)
                #6  0x000073e3af6aba37 _ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent (libQt5Core.so.5 + 0x2aba37)
                #7  0x000073e3af6f9a80 _ZN14QTimerInfoList14activateTimersEv (libQt5Core.so.5 + 0x2f9a80)
                #8  0x000073e3af6fa202 n/a (libQt5Core.so.5 + 0x2fa202)
                #9  0x000073e3afaa5a89 n/a (libglib-2.0.so.0 + 0x5ca89)
                #10 0x000073e3afb079b7 n/a (libglib-2.0.so.0 + 0xbe9b7)
                #11 0x000073e3afaa4f95 g_main_context_iteration (libglib-2.0.so.0 + 0x5bf95)
                #12 0x000073e3af6fa37f _ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE (libQt5Core.so.5 + 0x2fa37f)
                #13 0x000073e3af6a382c _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE (libQt5Core.so.5 + 0x2a382c)
                #14 0x000073e3af6afbfd _ZN16QCoreApplication4execEv (libQt5Core.so.5 + 0x2afbfd)
                #15 0x000055d47aa62c15 n/a (appimagelauncherd + 0xac15)
                #16 0x000073e3aee39c88 n/a (libc.so.6 + 0x25c88)
                #17 0x000073e3aee39d4c __libc_start_main (libc.so.6 + 0x25d4c)
                #18 0x000055d47aa635b5 n/a (appimagelauncherd + 0xb5b5)
                ELF object binary architecture: AMD x86-64
```

</p>
</details>

Just to show the extent of the problem, here is what I see every minute in `dmesg` logs:

<details><summary><code>sudo dmesg</code></summary>
<p>

```console
$ sudo dmesg | tail -n50

[Jul 9 23:08] traps: appimagelaunche[75543] general protection fault ip:7561614c6c04 sp:7ffce9ebeb58 error:0 in libstdc++.so.6.0.33[756161497000+14c000]
[Jul 9 23:09] traps: appimagelaunche[75603] general protection fault ip:7f90a88c6c04 sp:7ffc31aa0828 error:0 in libstdc++.so.6.0.33[7f90a8897000+14c000]
[ +39.999017] traps: appimagelaunche[75664] general protection fault ip:7fdf8f4c6c04 sp:7fffe513f238 error:0 in libstdc++.so.6.0.33[7fdf8f497000+14c000]
[Jul 9 23:10] traps: appimagelaunche[75705] general protection fault ip:7ecd2acc6c04 sp:7fffd0f98f98 error:0 in libstdc++.so.6.0.33[7ecd2ac97000+14c000]
[Jul 9 23:11] traps: appimagelaunche[75764] general protection fault ip:788a5d2c6c04 sp:7ffeeb8ef028 error:0 in libstdc++.so.6.0.33[788a5d297000+14c000]
[ +39.998929] traps: appimagelaunche[75824] general protection fault ip:74bf74ac6c04 sp:7ffe47b00f58 error:0 in libstdc++.so.6.0.33[74bf74a97000+14c000]
[Jul 9 23:12] traps: appimagelaunche[75865] general protection fault ip:7e369bcc6c04 sp:7ffee9ffdb28 error:0 in libstdc++.so.6.0.33[7e369bc97000+14c000]
[Jul 9 23:13] traps: appimagelaunche[75926] general protection fault ip:71c1432c6c04 sp:7fff0f292cb8 error:0 in libstdc++.so.6.0.33[71c143297000+14c000]
[ +39.999191] traps: appimagelaunche[75985] general protection fault ip:7b5e260c6c04 sp:7fff09280b88 error:0 in libstdc++.so.6.0.33[7b5e26097000+14c000]
[Jul 9 23:14] traps: appimagelaunche[76025] general protection fault ip:71e6a46c6c04 sp:7ffd7106e328 error:0 in libstdc++.so.6.0.33[71e6a4697000+14c000]
[Jul 9 23:15] traps: appimagelaunche[76084] general protection fault ip:71819f6c6c04 sp:7fffdc3db638 error:0 in libstdc++.so.6.0.33[71819f697000+14c000]
[ +39.999547] traps: appimagelaunche[76150] general protection fault ip:7163d66c6c04 sp:7ffd4fe95fe8 error:0 in libstdc++.so.6.0.33[7163d6697000+14c000]
...skipping...
[Jul10 07:47] traps: appimagelaunche[149675] general protection fault ip:7557fe2c6c04 sp:7ffdc13ec438 error:0 in libstdc++.so.6.0.33[7557fe297000+14c000]
[ +39.999654] traps: appimagelaunche[149739] general protection fault ip:776a24ec6c04 sp:7fffb25ddba8 error:0 in libstdc++.so.6.0.33[776a24e97000+14c000]
[Jul10 07:48] traps: appimagelaunche[149812] general protection fault ip:717f734c6c04 sp:7ffc936c3308 error:0 in libstdc++.so.6.0.33[717f73497000+14c000]
[Jul10 07:49] traps: appimagelaunche[149877] general protection fault ip:735c0d8c6c04 sp:7ffc6b6278d8 error:0 in libstdc++.so.6.0.33[735c0d897000+14c000]
[ +39.999506] traps: appimagelaunche[149938] general protection fault ip:78da504c6c04 sp:7ffe9740d028 error:0 in libstdc++.so.6.0.33[78da50497000+14c000]
[Jul10 07:50] traps: appimagelaunche[150048] general protection fault ip:7b934bac6c04 sp:7fff672a3ea8 error:0 in libstdc++.so.6.0.33[7b934ba97000+14c000]
[Jul10 07:51] traps: appimagelaunche[150112] general protection fault ip:7f8ea9cc6c04 sp:7ffcc2925fd8 error:0 in libstdc++.so.6.0.33[7f8ea9c97000+14c000]
[ +39.998806] traps: appimagelaunche[150173] general protection fault ip:7000e84c6c04 sp:7ffdb3dc9018 error:0 in libstdc++.so.6.0.33[7000e8497000+14c000]
[Jul10 07:52] traps: appimagelaunche[150245] general protection fault ip:7bdb774c6c04 sp:7ffe2bb8e688 error:0 in libstdc++.so.6.0.33[7bdb77497000+14c000]
[Jul10 07:53] traps: appimagelaunche[150309] general protection fault ip:750b7c6c6c04 sp:7ffef4812a18 error:0 in libstdc++.so.6.0.33[750b7c697000+14c000]
[ +39.999423] traps: appimagelaunche[150391] general protection fault ip:7e89a7ec6c04 sp:7ffceeda76f8 error:0 in libstdc++.so.6.0.33[7e89a7e97000+14c000]
[Jul10 07:54] traps: appimagelaunche[150460] general protection fault ip:7ec5abec6c04 sp:7ffda0080bb8 error:0 in libstdc++.so.6.0.33[7ec5abe97000+14c000]
[Jul10 07:55] traps: appimagelaunche[150527] general protection fault ip:76d6084c6c04 sp:7ffc083ae2a8 error:0 in libstdc++.so.6.0.33[76d608497000+14c000]
[ +39.999672] traps: appimagelaunche[150625] general protection fault ip:7b17cc4c6c04 sp:7ffd034a29a8 error:0 in libstdc++.so.6.0.33[7b17cc497000+14c000]
[Jul10 07:56] traps: appimagelaunche[150736] general protection fault ip:748e3c8c6c04 sp:7fff7053cb58 error:0 in libstdc++.so.6.0.33[748e3c897000+14c000]
[Jul10 07:57] traps: appimagelaunche[150796] general protection fault ip:735f11ac6c04 sp:7ffc05f12be8 error:0 in libstdc++.so.6.0.33[735f11a97000+14c000]
[ +39.998469] traps: appimagelaunche[150861] general protection fault ip:730a560c6c04 sp:7fffa2e9dfb8 error:0 in libstdc++.so.6.0.33[730a56097000+14c000]
[Jul10 07:58] traps: appimagelaunche[150930] general protection fault ip:79f44eec6c04 sp:7fffef2f00c8 error:0 in libstdc++.so.6.0.33[79f44ee97000+14c000]
[Jul10 07:59] traps: appimagelaunche[150996] general protection fault ip:75051f2c6c04 sp:7ffcbc693d08 error:0 in libstdc++.so.6.0.33[75051f297000+14c000]
[ +40.000001] traps: appimagelaunche[151058] general protection fault ip:7dbbed4c6c04 sp:7ffcef1886c8 error:0 in libstdc++.so.6.0.33[7dbbed497000+14c000]
[Jul10 08:00] traps: appimagelaunche[151128] general protection fault ip:6ffcc3cc6c04 sp:7ffd70d71c68 error:0 in libstdc++.so.6.0.33[6ffcc3c97000+14c000]
[Jul10 08:01] traps: appimagelaunche[151201] general protection fault ip:795e952c6c04 sp:7ffe86d52ca8 error:0 in libstdc++.so.6.0.33[795e95297000+14c000]
[ +39.999438] traps: appimagelaunche[151268] general protection fault ip:773e67cc6c04 sp:7ffdc0f33058 error:0 in libstdc++.so.6.0.33[773e67c97000+14c000]
[Jul10 08:02] traps: appimagelaunche[151328] general protection fault ip:764c188c6c04 sp:7ffd23d08018 error:0 in libstdc++.so.6.0.33[764c18897000+14c000]
[Jul10 08:03] traps: appimagelaunche[151405] general protection fault ip:7df1c8ec6c04 sp:7fff12ee7368 error:0 in libstdc++.so.6.0.33[7df1c8e97000+14c000]
[ +24.965518] traps: appimagelaunche[151467] general protection fault ip:7e1f630c6c04 sp:7fff9a980bd8 error:0 in libstdc++.so.6.0.33[7e1f63097000+14c000]
[Jul10 08:04] traps: appimagelaunche[151969] general protection fault ip:76e2e14c6c04 sp:7ffe4869d0c8 error:0 in libstdc++.so.6.0.33[76e2e1497000+14c000]
[Jul10 08:05] traps: appimagelaunche[152431] general protection fault ip:7dcb500c6c04 sp:7ffd40fdb7d8 error:0 in libstdc++.so.6.0.33[7dcb50097000+14c000]
[ +39.999592] traps: appimagelaunche[152776] general protection fault ip:7578280c6c04 sp:7ffebe2a0008 error:0 in libstdc++.so.6.0.33[757828097000+14c000]
[Jul10 08:06] traps: appimagelaunche[153020] general protection fault ip:717be80c6c04 sp:7ffd734bf488 error:0 in libstdc++.so.6.0.33[717be8097000+14c000]
[Jul10 08:07] traps: appimagelaunche[153549] general protection fault ip:7e74cccc6c04 sp:7ffcd1730a38 error:0 in libstdc++.so.6.0.33[7e74ccc97000+14c000]
[ +39.999338] traps: appimagelaunche[154452] general protection fault ip:7592226c6c04 sp:7ffe4c903f08 error:0 in libstdc++.so.6.0.33[759222697000+14c000]
[Jul10 08:08] traps: appimagelaunche[155035] general protection fault ip:79667fac6c04 sp:7ffc4b502e28 error:0 in libstdc++.so.6.0.33[79667fa97000+14c000]
[Jul10 08:09] traps: appimagelaunche[155586] general protection fault ip:7dbb506c6c04 sp:7ffc15f99198 error:0 in libstdc++.so.6.0.33[7dbb50697000+14c000]
[ +39.998462] traps: appimagelaunche[155820] general protection fault ip:7a4db38c6c04 sp:7ffd05a1d368 error:0 in libstdc++.so.6.0.33[7a4db3897000+14c000]
[Jul10 08:10] traps: appimagelaunche[155885] general protection fault ip:73e3af0c6c04 sp:7ffe465b6fa8 error:0 in libstdc++.so.6.0.33[73e3af097000+14c000]
[Jul10 10:18] perf: interrupt took too long (2518 > 2500), lowering kernel.perf_event_max_sample_rate to 79200
```

</p>
</details>

Seems that the user-level SystemD unit `appimagelauncherd.service` continually crashes each minute and gets restarted.

[1]: https://forum.manjaro.org/t/debug-symbols-on-manjaro/61157/13

**Comment by rcky844** [comment]
> `appimagelauncherd`

Just to report on this. I am having the same issue on 2.2.0-8.

**Comment by trinitronx** [comment]
@rcky844:
> Just to report on this. I am having the same issue since 2.2.0-8.

Yes, it's also still happening as of `extra/appimagelauncher` version `2.2.0-10`.

### TLDR; - workaround

To clean up only the oldest coredump files (default on Manjaro was 2 weeks old):

    systemd-tmpfiles --clean
    # OR:
    sudo systemctl start systemd-tmpfiles-clean.service

**Note:** The time limit is set by `tmpfiles.d` configs.  See [`man tmpfiles.d`][5] and: `grep -rin '/var/lib/systemd/coredump' /usr/lib/tmpfiles.d/`.

To clean up **all** the `appimagelauncher` coredump files:

    sudo rm -f /var/lib/systemd/coredump/core.appimagelaunche*

**Note:** This will leave dangling references to the old coredumps in JournalD (and thus `coredumpctl list`).  You'll notice that it says `missing` in the corefile column (or JSON field).  This is harmless and they will also eventually disappear when JournalD logs are expired or manually vacuumed (e.g. `sudo journalctl --vacuum-time=1d`).  Generally, the main problem to be solved is freeing up the disk space used by the coredump files.

To stop the `appimagelauncherd` SystemD service:

    systemctl disable --user appimagelauncherd.service

To disable the `appimagelauncherd` SystemD service:

    systemctl disable --user appimagelauncherd.service

To configure the maximum space used by coredumps stored "externally" (outside JournalD, e.g. on the filesystem), configure [`MaxUse` in `coredump.conf`][1].  For example, to set a `2` Gigabyte limit:

    [ -d /etc/systemd/coredump.conf.d/ ] || sudo mkdir  /etc/systemd/coredump.conf.d/
    echo 'MaxUse=2G' | sudo tee /etc/systemd/coredump.conf.d/01-coredump-maxuse.conf

**Note:** This will take effect only the next time a coredump is created. (see [`man systemd-coredump`][4]):

> A new instance of systemd-coredump is invoked upon receiving every core dump. Therefore, changes in these files will take effect the next time a core dump is received.

### Problem Details

Confirmed on Manjaro `unstable`, which is essentially like using latest Arch Linux rolling release, with `extra/appimagelauncher` version `2.2.0-10`:

```console
$ coredumpctl list --json=short | jq '.[] | select( .exe == "/usr/bin/appimagelauncherd" ) | .exe' | wc -l

6247

# Calculate total size of appimagelauncherd coredumps
$ find  /var/lib/systemd/coredump -type f -iname '*appimagelaunc*' -print0 | xargs -0 stat --format='%s' | awk '{s+=$1} END {printf "%.0f\n", s}'
1892293267

# Alternative method to find total size of these cordumps
$ coredumpctl list --json=short | jq -r '.[] | select( .exe == "/usr/bin/appimagelauncherd" and .size != null ) | .size '  | awk '{s+=$1} END {printf "%.0f\n", s}'
1892293267
```

So in my case, it was using `1892293267 bytes` or around `1.76 GiB`.  (In the past, before I realized about this issue, it grew much larger than this!)

This was on a longrunning system that had `appimagelauncherd.service` restarting due to SystemD's restart `on-failure` setting in `/usr/lib/systemd/user/appimagelauncherd.service`:

```
[Service]
ExecStart=/usr/bin/appimagelauncherd
Restart=on-failure
RestartSec=10
```

SystemD in this case keeps restarting the unit, which segfaults and crashes creating a coredump, then repeats after `RestartSec` (subject to any configured [`RestartSteps` backoff][3]).  Eventually all these coredump files build up, taking up space on disk until someone notices.

[1]: https://github.com/TheAssassin/AppImageLauncher/issues/558
[2]: https://www.freedesktop.org/software/systemd/man/latest/coredump.conf.html#MaxUse=
[3]: https://github.com/systemd/systemd/pull/26902
[4]: https://www.freedesktop.org/software/systemd/man/latest/systemd-coredump.html#Configuration
[5]: https://www.freedesktop.org/software/systemd/man/latest/tmpfiles.d.html#

**Comment by trinitronx** [comment]
@TheAssassin  I _think_ this is a duplicate of #333.  Since it's the older issue, I posted a [coredump over there][1].


P.S.: Sorry but the Arch Linux and thus Manjaro packages must've been built with debug symbols stripped for at least the `appimagelauncher` code.  I do see that `libstdc++.so.6` and a few other libraries like `libQt5Core.so.5` have some of them. Also, Arch Linux & Manjaro have made it difficult to get `debuginfod` working, [due to this][2].

I can probably try to build a custom package if you require more detailed debug symbols.  Based on [similar reports][3] from others (also [in #333][4])

[1]: https://github.com/TheAssassin/AppImageLauncher/issues/333#issuecomment-2351418680
[2]: https://forum.manjaro.org/t/debug-symbols-on-manjaro/61157/5?u=trinitronx
[3]: https://www.reddit.com/r/archlinux/comments/uyjz03/appimagelauncher_process_error_logs_are_clogging/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button
[4]: https://github.com/TheAssassin/AppImageLauncher/issues/333#issuecomment-791291079
